src/s3.zig

```diff
@@ -147,6 +225,46 @@ pub const AWSCredentials = struct {
                         new_credentials.options.queueSize = @intCast(@max(queueSize, std.math.maxInt(u8)));
                     }
                 }
+
+                if (try opts.getOptional(globalObject, "retry", i32)) |retry| {
+                    if (retry < 0 and retry > 255) {
+                        return globalObject.throwRangeError(retry, .{
+                            .min = 0,
+                            .max = 255,
+                            .field_name = "retry",
+                        });
+                    } else {
+                        new_credentials.options.retry = @intCast(retry);
+                    }
+                }
+
+                if (try opts.getTruthyComptime(globalObject, "acl")) |js_value| {
+                    if (!js_value.isEmptyOrUndefinedOrNull()) {
+                        if (js_value.isString()) {
+                            const str = bun.String.fromJS(js_value, globalObject);
```

Comment:
If you use `.getOptionalEnum` and then `ComptimeStringMap`, it will replace all this code and generate a message for you